tests/stress/km/stress_tests_km.cpp

```diff
@@ -693,127 +693,140 @@ _mt_prog_load_stress_test(ebpf_execution_type_t program_type, const test_control
-        context.thread_index,
-        map_fd,
-        context.map_name.c_str(),
-        context.file_name.c_str());
-    context.map_fd = map_fd;
-
-    // Get program object for (the only) program in this file.
+    // Get program object for the (only) program in this file.
     auto program = bpf_object__next_program(object_raw_ptr, nullptr);
     if (program == nullptr) {
         LOG_ERROR(
             "{}({}) FATAL ERROR: bpf_object__next_program({}) failed. errno:{}",
             __func__,
-            context.thread_index,
-            context.file_name.c_str(),
+            thread_index,
+            file_name.c_str(),
             errno);
         REQUIRE(program != nullptr);
     }
     LOG_VERBOSE(
         "{}({}) Found program object for program:{}, file_name:{}",
         __func__,
-        context.thread_index,
+        thread_index,
         program->program_name,
-        context.file_name.c_str());
+        file_name.c_str());
 
     // Get the fd for this program.
-    auto program_fd = bpf_program__fd(program);
+    fd_t program_fd = bpf_program__fd(program);
     if (program_fd < 0) {
         LOG_ERROR(
             "{}({}) FATAL ERROR: bpf_program__fd({}) failed. program:{}, errno:{}",
             __func__,
-            context.thread_index,
-            context.file_name.c_str(),
+            thread_index,
+            file_name.c_str(),
             program->program_name,
             errno);
         REQUIRE(program_fd >= 0);
     }
     LOG_VERBOSE(
         "{}({}) Opened fd:{}, for program:{}, file_name:{}",
         __func__,
-        context.thread_index,
+        thread_index,
         program_fd,
         program->program_name,
-        context.file_name.c_str());
-    context.program_fd = program_fd;
+        file_name.c_str());
 
     // Enforce the 'unspecified' compartment id. In the absence of additional compartments in the system, (as of now)
     // a non-existent id causes a kernel assert in netebpfext.sys debug builds.
-    context.compartment_id = UNSPECIFIED_COMPARTMENT_ID;
-    result = bpf_prog_attach(program_fd, context.compartment_id, BPF_CGROUP_INET4_CONNECT, 0);
+    result = bpf_prog_attach(program_fd, UNSPECIFIED_COMPARTMENT_ID, BPF_CGROUP_INET4_CONNECT, 0);
     if (result != 0) {
         LOG_ERROR(
             "{}({}) FATAL ERROR: bpf_prog_attach({}) failed. program:{}, errno:{}",
             __func__,
-            context.thread_index,
-            context.file_name.c_str(),
+            thread_index,
+            file_name.c_str(),
             program->program_name,
             errno);
         REQUIRE(result == 0);
     }
     LOG_VERBOSE(
-        "{}({}) Attached program:{}, file_name:{}",
-        __func__,
-        context.thread_index,
-        program->program_name,
-        context.file_name.c_str());
+        "{}({}) Attached program:{}, file_name:{}", __func__, thread_index, program->program_name, file_name.c_str());
+
+    object_ptr.reset(object_raw_ptr);
```

Comment:
Should this be done at `line 732` as that will ensure the fd's are closed in case of any failure in this function?

Reply:
fixed.